### ISSUE TITLE
E0010 needs to be updated to new format

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/rust#35194]
From: src/test/compile-fail/E0010.rs

A span_label is needed on E0010, taking it from:

```
error[E0010]: allocations are not allowed in constants
  --> src/test/compile-fail/E0010.rs:13:24
   |
13 | const CON : Box<i32> = box 0; //~ ERROR E0010
   |                        ^^^^^
```

To:

```
error[E0010]: allocations are not allowed in constants
  --> src/test/compile-fail/E0010.rs:13:24
   |
13 | const CON : Box<i32> = box 0; //~ ERROR E0010
   |                        ^^^^^ allocation not allowed in constant
```

**Comment by pcn** [comment]
Hi, I've got my attempt at updating E0010. It's based on my reading of http://www.jonathanturner.org/2016/08/helping-out-with-rust-errors.html, and it seems to output what I'd expect when I test it.

**Comment by KiChjang** [comment]
This should be fixed by #35439.
